subsys/net/lib/nrf_cloud/src/nrf_cloud_download.c

```diff
@@ -34,42 +39,99 @@ static K_MUTEX_DEFINE(active_dl_mutex);
 static struct nrf_cloud_download_data active_dl = { .type = NRF_CLOUD_DL_TYPE_NONE };
 
 #if defined(CONFIG_NRF_CLOUD_COAP_DOWNLOADS)
-/* CoAP client to be used for file downloads */
-static struct nrf_cloud_coap_client coap_client;
+
+const uint8_t coap_auth_request_template[] = {
+	0x48, 0x02, /* CoAP POST */
+	0xFF, 0xFF, /* Message ID */
+	0x90, 0x4A, 0x50, 0x61, 0xBF, 0x40, 0x33, 0x40, /* Token */
+	0xB4, 0x61, 0x75, 0x74, 0x68, /* URI: "auth" */
+	0x03, 0x6A, 0x77, 0x74, /* URI "jwt" */
+	0x10, /* Content-Type Text */
+	0xFF, /* Payload-Marker */
+};
+#define JWT_BUF_SZ 600
+#define COAP_AUTH_REQ_HEADER_SIZE (sizeof(coap_auth_request_template))
+#define COAP_AUTH_REQ_BUF_SIZE (JWT_BUF_SZ + COAP_AUTH_REQ_HEADER_SIZE)
 
 int nrf_cloud_download_handle_coap_auth(int socket)
 {
-	int err = 0;
-	struct nrf_cloud_coap_client *client = &coap_client;
-
-	/* Initialize client */
-	err = nrf_cloud_coap_transport_init(client);
+	int32_t err = 0;
+	struct coap_packet reply;
+	uint8_t token[COAP_TOKEN_MAX_LEN];
+	uint8_t *packet_buf = nrf_cloud_malloc(COAP_AUTH_REQ_BUF_SIZE);
+	uint8_t response_token[COAP_TOKEN_MAX_LEN];
+	uint16_t mid = sys_cpu_to_be16(coap_next_id());
+	size_t request_size;
+
+	/* Make new random token */
+	sys_rand_get(token, COAP_TOKEN_MAX_LEN);
+
+	/* Fill in CoAP header */
+	memcpy(packet_buf, coap_auth_request_template, COAP_AUTH_REQ_HEADER_SIZE);
+	memcpy(packet_buf + 2, &mid, sizeof(mid));
+	memcpy(packet_buf + 4, token, COAP_TOKEN_MAX_LEN);
```

Comment:
use defines or sizeof() instead of magic numbers

Reply:
I put them in defines now